File: TRA Task/problem 3.java
import java.util.Scanner;

public class task3 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        boolean flag = true;

        while(flag ){
            System.out.println("Menu:");
            System.out.println("1. Display a right-angle triangle of ones");
            System.out.println("2. Display a Palindromic Triangle");
            System.out.println("3. Help");
            System.out.println("4. Exit");
         
            System.out.print("Enter your choice: ");
            int choice = sc.nextInt();
            
            // right-angle triangle
            if (choice == 1) {
                System.out.print("Enter the number of lines: ");
                int rows = sc.nextInt();

                for(int i = 0; i < rows; i++){
                    for(int j = 0; j <= i; j++){
                        System.out.print("1");
                    }
                    System.out.println();
                }

                choice = 0;
            }
            
            // Palindromic Triangle
            else if (choice == 2 ){
                System.out.print("Enter the number of lines: ");
                int rows = sc.nextInt();

                for(int i = 1; i <= rows; i++){
                    for (int j = 1; j <= i; j++)
                        System.out.print(j);
                    for (int j = i - 1; j >= 1; j--)
                        System.out.print(j);

                    System.out.println();
                }

                choice = 0;
            }
            // Help
            else if (choice == 3 ){
                System.out.println("Help:");
        System.out.println("A Palindromic Triangle is a triangular array of numbers where each row forms a palindromic.");
        System.out.println("The first new lines of palindromic Triangule are: ");
                System.out.println("1");
                System.out.println("121");
                System.out.println("12321");
                System.out.println("1234321");
         System.out.println("You can use this pattern to draw a palindromic triangle for any number of lines. ");
            }
            // Exit
            else if (choice == 4 ){
                flag = false;
                System.out.println("Exiting the program.");
            }


        }
    }
}

File: problem-3.py
def right_angle_triangle(lines):
    for i in range(1, lines + 1):
        print('1 ' * i)

def palindromic_triangle(lines):
    for i in range(1, lines + 1):
        for j in range(1, i + 1):
            print(j, end=' ')
        for j in range(i - 1, 0, -1):
            print(j, end=' ')
        print()

def display_help():
    print("\nHelp:")
    print("1. Right-angle Triangle of Ones:")
    print("   This option displays a right-angled triangle made of '1's.")
    print("   The number of lines determines the height of the triangle.")
    print("\n2. Palindromic Triangle:")
    print("   This option displays a triangle with palindromic number patterns.")
    print("   Each line is a palindrome, and the number of lines determines")
    print("   the size of the triangle.")

def main():
    while True:
        print("\nMenu:")
        print("1. Display a right-angle triangle of ones")
        print("2. Display a palindromic triangle")
        print("3. Help")
        print("4. Exit")
        
        choice = input("Enter your choice (1-4): ")
        
        if choice == '1':
            lines = int(input("Enter the number of lines: "))
            right_angle_triangle(lines)
        elif choice == '2':
            lines = int(input("Enter the number of lines: "))
            palindromic_triangle(lines)
        elif choice == '3':
            display_help()
        elif choice == '4':
            print("Exiting the program. Goodbye!")
            break
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()